DogScepterLib/Core/GMData.cs

```diff
@@ -4,109 +4,195 @@
-                }
-                else if (Major == major)
-                {
-                    if (Minor < minor)
-                    {
-                        Minor = minor;
-                        Release = release;
-                        Build = build;
-                    }
-                    else if (Minor == minor)
-                    {
-                        if (Release < release)
-                        {
-                            Release = release;
-                            Build = build;
-                        }
-                        else if (Release == release)
-                        {
-                            if (Build < build)
-                                Build = build;
-                        }
-                    }
-                }
+                Minor = minor;
+                Release = release;
+                Build = build;
+                return;
             }
+            if (Minor > minor) return;
 
-            /// <summary>
-            /// Returns whether the recorded version number is greater than or equal to the parameters
-            /// </summary>
-            public bool IsNumberAtLeast(int major = 0, int minor = 0, int release = 0, int build = 0)
+            // Our Minor is now equal to proposed minor.
+            if (Release < release)
             {
-                if (Major != major)
-                    return (Major > major);
-                if (Minor != minor)
-                    return (Minor > minor);
-                if (Release != release)
-                    return (Release > release);
-                if (Build != build)
-                    return (Build > build);
-                return true;
+                Release = release;
+                Build = build;
+                return;
             }
+            if (Release > release) return;
+
+            // Our Release is now equal to proposed release.
+            if (Build < build)
+                Build = build;
         }
 
-        public class GMStats
+        /// <summary>
+        /// Returns whether the recorded version number is greater than or equal to the parameters.
+        /// </summary>
+        public bool IsNumberAtLeast(int major = 0, int minor = 0, int release = 0, int build = 0)
         {
-            public int LastStructID = -1;
+            if (Major != major)
+                return (Major > major);
+            if (Minor != minor)
+                return (Minor > minor);
+            if (Release != release)
+                return (Release > release);
+            if (Build != build)
+                return (Build > build);
+            return true;
         }
+    }
 
-        public GMVersionInfo VersionInfo;
-        public GMStats Stats;
-
-        public GMChunkFORM FORM;
-        public Dictionary<string, GMChunk> Chunks => FORM.Chunks;
-#if DEBUG
-        public Log Logger = Console.WriteLine;
-#else
-        public Log Logger = null;
-#endif
+    //TODO:?
```

Comment:
Right now this class is solely used to track the last GML struct ID (for 2.3+ games) so that the compiler knows what ID to produce next. It may be populated with more info in the future, potentially with cosmetic fields as well (like general stats about the data file that could be interesting, but maybe not).

Reply:
So essentially, stats generated from / are for DS, and not contained directly within the data file itself?

Reply:
Yes